components/styled/table/src/molecules/table-core.tsx

```diff
@@ -0,0 +1,148 @@
+import styled from '@emotion/styled'
+import { Icon, IconButton } from '@ltht-react/icon'
+import { TABLE_COLOURS, TEXT_COLOURS } from '@ltht-react/styles'
+import { flexRender, Table } from '@tanstack/react-table'
+import { Dispatch, SetStateAction } from 'react'
+import { CellProps } from './table-cell'
+import { calculateStaticColumnOffset } from './table-methods'
+import { StandardButtonStyle, StyledTableData, StyledTableHeader } from './table-styles'
+
+const DefaultTableOptions = {
+  showExpanderColumn: false,
+  pageSize: 10,
+  enablePagination: false,
+  serverSidePagination: false,
+  hidePaginationControls: false,
+  hidePerPageOptions: false,
+}
+
+const DefaultPerPageOptions = [10, 20, 30, 40, 50]
+
+const buildTableHead = (
```

Comment:
If all these are doing is abstracting some JSX, could we make them functional components instead?